immutable_tree.go

```diff
@@ -38,6 +38,50 @@ func (t *ImmutableTree) String() string {
 	return "Tree{" + strings.Join(leaves, ", ") + "}"
 }
 
+// RenderShape provides a nested tree shape, ident is prepended in each level
+// Returns an array of strings, one per line, to join with "\n" or display otherwise
+func (t *ImmutableTree) RenderShape(indent string, encoder func([]byte, int, bool) string) []string {
```

Comment:
At least add a comment about the `encoder`. Alternatively, introduce a commented type alias. Something along the lines of: `type NodeEncoder func(id []byte, depth int, isLeaf bool) string` or an interface:
```go
type NodeEncoder interface {
	Encode(id []byte, depth int, isLeaf bool) string
}
```

Reply:
Will do.